File: Data Structures & Algorithms/sort-colors/submission-0.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """

        l, i, r = 0, 0, len(nums) - 1

        while i <= r:
            if nums[i] == 0:
                nums[l], nums[i] = nums[i], nums[l]
                l += 1
                i += 1
            elif nums[i] == 2:
                nums[r], nums[i] = nums[i], nums[r]
                r -= 1
                # Do NOT increment i here, because swapped element needs to be checked
            else:  # nums[i] == 1
                i += 1

            
        

File: Data Structures & Algorithms/concatenation-of-array/submission-0.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        
        int i = 0;
        int r = nums.length;
        int [] ans = new int[2*r];
        while(i< r){
            ans[i] = nums[i];
            ans[i+r] = nums[i];
             i += 1;
        }
        return ans;
        
        
    }
}